src/webgldrawer.js

```diff
@@ -379,23 +376,13 @@
                         let tile = tilesToDraw[tileIndex].tile;
                         let indexInDrawArray = tileIndex % maxTextures;
                         let numTilesToDraw =  indexInDrawArray + 1;
-                        let tileContext = tile.getCanvasContext();
-
-                        let textureInfo = tileContext ? this._TextureMap.get(tileContext.canvas) : null;
-                        if(!textureInfo){
-                            // tile was not processed in the tile-ready event (this can happen
-                            // if this drawer was created after the tile was downloaded)
-                            this._tileReadyHandler({tile: tile, tiledImage: tiledImage});
 
-                            // retry getting textureInfo
-                            textureInfo = tileContext ? this._TextureMap.get(tileContext.canvas) : null;
+                        const textureInfo = this.getDataToDraw(tile);
+                        if (!textureInfo) {
+                            return;
                         }
+                        this._getTileData(tile, tiledImage, textureInfo, overallMatrix, indexInDrawArray, texturePositionArray, textureDataArray, matrixArray, opacityArray);
```

Comment:
I do not agree with changing the guard approach here. By returning, you introduce the possibility of inappropriately skipping the steps below to render any tiles in the pipeline. We don't want to do that.

Reply:
This is not how it works. `getDataToDraw` is a method that returns data for the current tile. If it returns a falsey value (undefined), it tells you the tile data is not ready for processing, the return condition here is just a safe escape from invalid state of the drawer. By design, this condition should never happen, unless there is a bug. And if there is a bug, `getDataToDraw` will print a warning somewhere internally. All drawers are supposed to do this.

Reply:
Sure, you can't draw the data for a tile that doesn't have data... I'm not arguing that you should do that.

But the loop this is within doesn't deal *only* with the current tile. On the line linked below, it checks whether to draw all the tiles in the buffer:

https://github.com/openseadragon/openseadragon/blob/f03f2a5d313ca98dfc0f77472eafa0c8e1c21f7f/src/webgldrawer.js#L386

With the way the guard is currently implemented, you increase a counter above the guard, then possibly skip the condition where the buffer would be flushed. If you're unlucky, that was the loop on which the buffer should be drawn and reset; then you'd cause a buffer overflow on the next loop, and you'd really be causing a problem.

The old logic worked fine. The change to `return`ing from the loop is actually introducing the possibility of causing not just an invalid tile state but causing the entire drawer to crash.

Reply:
Aha, there should be 'continue'. My bad, thanks for noticing.